### PR TITLE
#424 validate name existed

### PR DESCRIPTION
**Pull request by DinhThaiVV**

### CODE REVIEW

**Review by LEHOANGGLAM — APPROVED** [review]

File: product/src/main/java/com/yas/product/repository/BrandRepository.java [hunk 1]
```diff
@@ -2,11 +2,15 @@
...
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.stereotype.Repository;
 
 import java.util.Optional;
 
 @Repository
 public interface BrandRepository extends JpaRepository<Brand,Long> {
     Optional<Brand> findBySlug(String slug);
+
+    @Query("select e from Brand e where e.name = ?1 and (?2 is null or e.id != ?2)")
```
Comment (toannguyenk) [inline product/src/main/java/com/yas/product/repository/BrandRepository.java:14]: find by Id but i see e.id != ?2
please check

File: product/src/main/java/com/yas/product/repository/CategoryRepository.java [hunk 2]
```diff
@@ -2,11 +2,15 @@
...
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.stereotype.Repository;
 
 import java.util.Optional;
 
 @Repository
 public interface CategoryRepository extends JpaRepository<Category, Long> {
     Optional<Category> findBySlug(String slug);
+
+    @Query("select e from Category e where e.name = ?1 and (?2 is null or e.id != ?2)")
```
Comment (toannguyenk) [inline product/src/main/java/com/yas/product/repository/CategoryRepository.java:14]: same here

**Review by lnghia — APPROVED** [review]

**Review by nguyenvanhadncntt — APPROVED** [review]

**Review by toannguyenk — APPROVED** [review]

File: product/src/main/java/com/yas/product/repository/CategoryRepository.java (hunk 2 above)
Comment (DinhThaiVV) [reply product/src/main/java/com/yas/product/repository/CategoryRepository.java:14]: Thanks Toan. I will rename method to clearly logic

File: product/src/main/java/com/yas/product/repository/BrandRepository.java (hunk 1 above)
Comment (DinhThaiVV) [reply product/src/main/java/com/yas/product/repository/BrandRepository.java:14]: Thanks Toan. I will rename method to clearly logic

**Pull request merged**